middleware/proxy.go

```diff
@@ -231,7 +236,17 @@ func ProxyWithConfig(config ProxyConfig) echo.MiddlewareFunc {
 
 			req := c.Request()
 			res := c.Response()
-			tgt := config.Balancer.Next(c)
+
+			var tgt *ProxyTarget
+			provider, ok := config.Balancer.(TargetProvider)
```

Comment:
this cast could be moved to line 230  just before `return func(next echo.HandlerFunc) echo.HandlerFunc {` so we do not need cast every request.